src/plugins/intel_cpu/src/nodes/scatter_update.cpp

```diff
@@ -263,6 +260,462 @@ static std::vector<size_t> getBlockND(const VectorDims& shape) {
+                    auto dst = dataPtr + (offsets[0] + idxValue * dataBlockND[axis + 1]);
+                    auto src = updatePtr + indices_offset;
+                    kernel_func(dst, src);
+                    indices_offset += indicesBlockND[axis + 1];
+                }
+                // increment
+                tensorItr.increment(offsets, dataBlockND, indicesBlockND);
+            }
+        } else {
+            // For better performance, the offsets of dst and indices are cached in the first iteration of outer loop, and reused
+            // in the remaining iterations.
+            std::vector<size_t> dst_offsets(end-start+1, offsets[0]);  // one extra to avoid overflow at the last iteration of inner loop
+            std::vector<size_t> indices_offsets(end-start+1, offsets[1]);
+            size_t *ptr_dst_offset = &dst_offsets[0];
+            size_t *ptr_indices_offset = &indices_offsets[0];
+            for (size_t worker = start; worker < end; worker++) { // idx = 0
+                IndexType idxValue = *(indicesPtr + *ptr_indices_offset);
+                if (idxValue < 0) idxValue += data_dim_size;
+                ASSERT_DEBUG_ONLY(idxValue < data_dim_size && idxValue >= 0, "invalid index value.");
+                auto dst = dataPtr + (*ptr_dst_offset + idxValue * dataBlockND[axis + 1]);
+                auto src = updatePtr + *ptr_indices_offset;
+                kernel_func(dst, src);
+
+                // increment once for all
+                tensorItr.increment(offsets, dataBlockND, indicesBlockND);
+                *++ptr_dst_offset = offsets[0];
+                *++ptr_indices_offset = offsets[1];
+            }
+            for (size_t idx = 1; idx < index_dim_size; idx++) {
+                ptr_indices_offset = &indices_offsets[0];
+                ptr_dst_offset = &dst_offsets[0];
+                for (size_t worker = start; worker < end; worker++) {
+                    auto indices_offset = *ptr_indices_offset + idx * indicesBlockND[axis + 1];
+                    IndexType idxValue = *(indicesPtr + indices_offset);
+                    if (idxValue < 0) idxValue += data_dim_size;
+                    ASSERT_DEBUG_ONLY(idxValue < data_dim_size && idxValue >= 0, "invalid index value.");
+                    auto dst = dataPtr + (*ptr_dst_offset + idxValue * dataBlockND[axis + 1]);
+                    auto src = updatePtr + indices_offset;
+                    kernel_func(dst, src);
+                    ptr_indices_offset++;
+                    ptr_dst_offset++;
+                }
+            }
+        }
+    });
+}
+
+// We specialize ReduceMean to avoid spoil performance of other reduce methods, as otherwise condition branch
+// were used in loops.
+template <typename DataType, typename IndexType>
+void scatterElementsUpdate(const MemoryPtr& mem_data, const MemoryPtr& mem_indices, const MemoryPtr& mem_updates,
```

Comment:
Just for the record, this specialization can be avoided considering following tech:
 - https://stackoverflow.com/questions/37617677/implementing-a-compile-time-static-if-logic-for-different-string-types-in-a-co
 - https://stackoverflow.com/questions/12160765/if-else-at-compile-time-in-c

Reply:
constexpr? sounds not available for c++11.